src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4564,14 +4564,34 @@ public void testRemoveService() throws Exception {
   @Test
   public void testInspectTask() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "swarm support");
+    final Date start = new Date();
 
-    final ServiceSpec spec = createServiceSpec(randomName());
+    final ServiceSpec serviceSpec = createServiceSpec(randomName());
     assertThat(sut.listTasks().size(), is(0));
-    sut.createService(spec);
+    final ServiceCreateResponse serviceCreateResponse = sut.createService(serviceSpec);
     await().until(numberOfTasks(sut), is(greaterThan(0)));
-    final Task task = sut.listTasks().get(0);
-    final Task inspectTask = sut.inspectTask(task.id());
-    assertThat(task, equalTo(inspectTask));
+
+    final Task someTask = sut.listTasks().get(0);
+    final Task inspectedTask = sut.inspectTask(someTask.id());
+    assertThat(inspectedTask.id(), notNullValue());
+    assertThat(inspectedTask.version().index(), allOf(notNullValue(), greaterThan(0L)));
+    final Date now = new Date();
+    assertThat(inspectedTask.createdAt(), allOf(notNullValue(), greaterThan(start), lessThan(now)));
+    assertThat(inspectedTask.updatedAt(), allOf(notNullValue(), greaterThan(start), lessThan(now)));
+    assertThat(inspectedTask.slot(), allOf(notNullValue(), greaterThan(0)));
+    assertThat(inspectedTask.status(), notNullValue());
+    assertThat(inspectedTask.name(), nullValue());
+    assertEquals(serviceCreateResponse.id(), inspectedTask.serviceId());
+    assertEquals(serviceSpec.taskTemplate(), inspectedTask.spec());
+    if (serviceSpec.labels() == null || serviceSpec.labels().isEmpty()) {
```

Comment:
I tried both of those. First problem: `empty()` only works for `Collection`s, not `Map`s. Hamcrest does have [`IsMapWithSize#anEmptyMap`](https://github.com/hamcrest/JavaHamcrest/blob/142c8a95d3bd4dddaef1bb09eb6875c56646f7a3/hamcrest-library/src/main/java/org/hamcrest/collection/IsMapWithSize.java#L57) but it is not in Hamcrest 1.3.

Even for lists, though, it is not actually easy to combine checks for `null` or `empty`. My IDE complains about both `either( empty() ).or( nullValue() )` and `anyOf( empty(), nullValue() )` because Hamcrest's generics are weird. There is probably a way to get those to work by specifying the classes in some `<>` diamonds, but I don't know how to do it.